socorro/external/es/super_search_fields.py

```diff
@@ -142,6 +140,23 @@ def add_field_to_properties(properties, namespaces, field):
             if not field.get('storage_mapping'):
                 continue
 
+            storage_mapping = field['storage_mapping']
+
+            # Add "doc_values": True to any non-string field or string field that
+            # doesn't have an analyzer
+            # NOTE(willkg): Elasticsearch 2.0+ does this automatically, so we
+            # can nix this.
+            if storage_mapping.get('fields'):
+                for subfield in storage_mapping.get('fields', {}).keys():
+                    field_mapping = storage_mapping['fields'][subfield]
+                    if ((field_mapping.get('type') != 'string' or
+                         field_mapping.get('index') == 'not_analyzed')):
+                        field_mapping['doc_values'] = True
+
+            if ((storage_mapping.get('type') != 'string' or
+                 storage_mapping.get('index') == 'not_analyzed')):
+                storage_mapping['doc_values'] = True
```

Comment:
This block is the meat of the change. We want to change the mapping for fields to add a `doc_values=True` iff:

1. the field is not a string type, OR
2. the field is a string type, but explicitly has `index=not_analyzed`

Fields can be nested. So for example, we have a field `abort_message` and we store an analyzed version of it, but also an unanalyzed version of it. We use the two for different search scenarios.

Given that fields can be nested (only one level of nesting), we want to make sure we get all the subfields.

This isn't glamorous code. Elasticsearch 2.0 switches the default from `doc_values=False` to `doc_values=True`. When/if we upgrade to something >= 2.0, we can nix this code because we won't need it anymore.